src/admm.jl

```diff
@@ -0,0 +1,139 @@
+"""
+        admm(
+            m::AbstractBlockNLPModel;
+            kwargs... 
+        )
+Solves a `BlockNLPModel` with the ADMM algorithm.
+
+# Arguments
+
+- `m::AbstractBlockNLPModel`: identifier for the `BlockNLPModel` 
+- `kwargs...`: options for the solver
+"""
+function admm(
+    m::AbstractBlockNLPModel;
+    option_dict::Dict{Symbol,Any} = Dict{Symbol,Any}(),
+    kwargs...,
+)
+    start_time = time()
+    nb = m.problem_size.block_counter # Number of blocks
+    iter_count = 0
+
+    # check if warm start primal-dual solutions are available 
+    # otherwise initialize them with zero vectors
+    if :primal_start ∈ keys(kwargs)
+        x = Vector{Float64}(kwargs[:primal_start])
```

Comment:
I am not sure to understand this line. `kwargs[:primal_start]` is a vector right? Why using `Vector{Float64}` there?

Reply:
I am just trying to make sure that the `primal_start` is always a vector of floats. In case the user enters it as a vector of integers, this step will convert it into floats. Do you think this is not required?

Reply:
In that case I think it's better to use the function `convert`. But let's address this is in a future PR.